build.sh

```diff
@@ -1,25 +1,25 @@
 #!/bin/bash
 set -e -x
-PREFIX=_build/prod/
-APP=api_web
-BUILD_TAG=$APP:_build
-VERSION=$(grep -o 'version: .*"' apps/$APP/mix.exs | grep -E -o '([0-9]+\.)+[0-9]+')
-BUILD_ARTIFACT=$APP-build.zip
 
-docker build -t $BUILD_TAG .
-CONTAINER=$(docker run -d ${BUILD_TAG} sleep 2000)
+# other required configuration:
+# * APP
+# * DOCKER_REPO
+
+BUILD_ARTIFACT=$APP-build.zip
 
-rm -rf rel/$APP rel/$APP.tar.gz
-docker cp $CONTAINER:/root/${PREFIX}rel/$APP/releases/$VERSION/$APP.tar.gz rel/$APP.tar.gz
+semaphore/build_push.sh
 
-docker kill $CONTAINER
-docker rm $CONTAINER
-tar -zxf rel/$APP.tar.gz -C rel/
-rm rel/$APP.tar.gz
+# Create Dockerrun file pointing to ECR image, and zip it up
+githash=$(git rev-parse --short HEAD)
+# escape slashes in the DOCKER_REPO
+repo=$(echo $DOCKER_REPO | sed "s/\//\\\\\//g")
 
-test -f $BUILD_ARTIFACT && rm $BUILD_ARTIFACT
-pushd rel
-zip -r ../$BUILD_ARTIFACT * .ebextensions
-rm -r bin erts* lib releases
-git checkout -- bin
+test -d $githash && rm -r $githash
+mkdir $githash
+sed -e "s/DOCKER_REPO/$repo/g" -e "s/GITHASH/$githash/g" ./semaphore/Dockerrun.aws.json > $githash/Dockerrun.aws.json
+cp -r rel/.ebextensions $githash
+pushd $githash
+zip -r ../$BUILD_ARTIFACT .
 popd
+
+rm -r $githash
```

Comment:
I was confused at first since the script here seemed to end prematurely, but then after poking around in semaphore I see that there's actually a fair bit of bash scripting there.

Does it make sense as part of this deployment overhaul to pull in some of that code to version control, either as part of this script, or another script `deploy.sh`? (Arrow, e.g., has two scripts, `build_push.sh`, which is the same as the one you added here, and `deploy.sh`, which is basically this file plus a lot of the stuff that is in semaphore right now.)

Reply:
In general, yes. However, in this case I wanted to be able to do this branch without needing to make changes to the Semaphore build/deploy configuration.